reactome_analysis_api/reactome_analysis_api/searcher/public_data_searcher.py

```diff
@@ -140,17 +140,33 @@ def index_search(self, keyword: list, species: str = None, search_in_description
             results_list = list()
             for result in results:
                 if result["id"] != '':
+                    web_link = self._add_link(result["resource_id"], result["id"])
                     results_list.append({
                         "id": result["id"], 
                         "description": result["description"],
                         "title": result["title"],
                         "species": result["species"],
                         "resource_id": result["resource_id"], 
                         "loading_parameters": result["loading_parameters"],
-                        "data_source": result["data_source"]
+                        "data_source": result["data_source"],
+                        "web_link": web_link
                     })
             return results_list
         
+    def _add_link(self, ressource_id: str, dataset_id: str) -> str:
```

Comment:
@alexgrent 
I'm just thinking whether this is the smartest way to do it. With this, if we add a new resource, we always also have to update this function.

Why not add this to the functions in `PublicDataFetcher`? There, we already have resource specific code anyway and if we add a new resource, we just add this function - with the links directly available in the object.

Reply:
changed acc. to review, in this case the link will also be stored in the index.